Pod/Classes/Playback/public/WistiaPlayer.swift

```diff
@@ -82,13 +82,24 @@ public protocol WistiaPlayerDelegate : class {
      that will always be the one used.  Otherwise, heuristics will use the current value of `requireHLS` and
      device characteristics to select the best asset.
      
+     - Parameter player: The `WistiaPlayer` loading the asset.
      - Parameter media: The `WistiaMedia` from which this asset was chosen.
      - Parameter asset: The `WistiaAsset` that will be loaded for playback, if the HLS master index manifest was not used.
      - Parameter usingHLSMasterIndexManifest: `True` iff playback is using the HLS master index manifest 
         (aka Manifest of Manifests). `asset` will be `nil`.
 
      */
     func wistiaPlayer(_ player:WistiaPlayer, willLoadVideoForMedia media:WistiaMedia, usingAsset asset:WistiaAsset?, usingHLSMasterIndexManifest: Bool)
+    
+    /**
+     Informs the delegate that the `WistiaPlayer` is attempting to determine a URL for loading a `WistiaAsset`, giving an opportunity
+     for the delegate to inject a url (such as a cached local asset url). The `WistiaPlayer` will use the URL returned by this method
+     for playback. Return nil to allow the `WistiaPlayer` to determine the best playback URL.
+     
+     - Parameter player: The `WistiaPlayer` determining the URL for playback.
+     - Parameter media: The 'WistiaMedia' from which a playback URL is needed.
+     */
+    func wistiaPlayer(_ player:WistiaPlayer, shouldLoadVideoForMedia media:WistiaMedia) -> URL?
 }
 
```

Comment:
If we add the following protocol extension here, current users of this protocol (including `WistiaPlayerViewController` within WK) wouldn't need to make any changes and wouldn't be affected.

```
public extension WistiaPlayerDelegate {

    //Default implementation retains standard functionality
    public func wistiaPlayer(_ player: WistiaPlayer, shouldLoadVideoForMedia media: WistiaMedia) -> URL? {
        return nil
    }

}
```